tensorflow/core/grappler/optimizers/remapper.cc

```diff
@@ -970,6 +982,123 @@ bool FindFusedBatchNormEx(const RemapperContext& ctx, int node_index,
   return false;
 }
 
+bool FindFusedBatchNormGradEx(const RemapperContext& ctx, int node_index,
+                              FusedBatchNormGradEx* matched) {
+  // Root of the pattern must be a FusedBatchNormGrad.
+  const auto* node_view = ctx.graph_view.GetNode(node_index);
+
+	// Returns true iff the node is a compatible FusedBatchNormGrad node.
+  const auto valid_batch_norm_grad =
+      [&](const utils::MutableNodeView& fused_batch_norm_grad) -> bool {
+    const auto* node_def = fused_batch_norm_grad.node();
+    if (!IsFusedBatchNormGrad(*node_def) ||
+        HasControlFaninOrFanout(fused_batch_norm_grad)) return false;
+
+    // We fuse FusedBatchNormGrad on GPU.
+    if (!NodeIsOnGpu(node_def)) return false;
+
+    // Data type must be DT_HALF.
+    DataType t_dtype = GetDataTypeFromAttr(*node_def, "T");
+    if (t_dtype != DT_HALF) return false;
+
+    // We rely on cuDNN for computing FusedBatchNormGrad with side
+    // outputs and activation. cuDNN only supports NHWC data layout.
+    string data_format;
+    if (!GetNodeAttr(*node_def, kDataFormat, &data_format).ok()) return false;
+    if (data_format != "NHWC") return false;
+
+    // Channel dimension must be a multiple of 4.
+    const auto& props = ctx.graph_properties.GetInputProperties(
+        node_def->name());
+    const bool valid_channel_dim = !props.empty() &&
+                                   props[0].shape().dim_size() == 4 &&
+                                   props[0].shape().dim(3).size() % 4 == 0;
+    if (!valid_channel_dim) return false;
+
+    // cuDNN must support CUDNN_BATCHNORM_SPATIAL_PERSISTENT mode.
+    if (!BatchnormSpatialPersistentEnabled()) return false;
+
+    // FusedBatchNormV2 and V3 have an extra type parameter.
+    if (node_def->op() != "FusedBatchNorm" &&
+        !HasDataType(node_def, DT_FLOAT, "U")) return false;
+
+    return true;
+  };
+
+  if (ctx.xla_auto_clustering_on) return false;
```

Comment:
This is not enough though: there are other ways to enable XLA. Additionally, we would be to dig to the bottom of the issue and figure out why the convolution configuration was marked as unsupported and whether it's specific to XLA.

Reply:
I agree. The reason I put a line here is that I heard the XLA might not be able to correctly process the new fused op. So, I tried to avoid remapping this pattern when XLA is on. For the "unsupported cudnn conv" issue, if you can share with more details on how to repro the failed model benchmark, I can look into it.

Reply:
Right, but still action item here is to have a more robust check. In Python, you could check whether you are inside XLA context, I'm not 100% sure what is the best analogue in Grappler.